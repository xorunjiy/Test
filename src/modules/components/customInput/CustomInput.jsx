import React from 'react';
import { Wrapper } from './styledComponents.js';
import PropTypes from 'prop-types';
import theme from '../../themes/colors.js'//принимать пропсой

const CustomInput = props => {
    const {
        type,
        name,
        label,
        width,
        value,
        height,
        display,
        checked,
        required,
        fontSize,
        labelColor,
        handleChange,
        inputBackground,
    } = props;

    return(
        <Wrapper display={display}>
            {label && 
            <Wrapper.label fontSize={fontSize}
                                children={label}
                                labelColor={labelColor}
            />}
            <Wrapper.input
                type={type}
                name={name}
                value={value}
                width={width}
                height={height}
                checked={checked}
                fontSize={fontSize}
                required={required}
                onChange={handleChange}
                inputBackground={inputBackground}
            />
        </Wrapper>
    )
}

CustomInput.propTypes = {
    type: PropTypes.string,
    label: PropTypes.string,
    width: PropTypes.number,
    value: PropTypes.string,
    height: PropTypes.number,
    checked: PropTypes.bool,
    display: PropTypes.string,
    required: PropTypes.bool,
    fontSize: PropTypes.number,
    labelColor: PropTypes.string,
    inputTitle: PropTypes.string,
    handleChange: PropTypes.func,
    inputBackground: PropTypes.string,

}

CustomInput.defaultProps = {
    type: 'text',
    width: 40,
    height: 25,
    display: 'block',
    fontSize: 18,
    required: false,
    value:'',
    inputBackground: theme.infoBackground,
}

export default React.memo(CustomInput);