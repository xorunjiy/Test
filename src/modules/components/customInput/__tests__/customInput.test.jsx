import React from 'react';
import CustomInput from '../CustomInput.jsx';

const props = {
    type: 'text',
    width: 40,
    height: 25,
}

describe('CustomInput snapshot', () => {
    it('should correctly render', () => {

        const component = renderer(<CustomInput {...props}/>);

        expect(component).toMatchSnapshot();
    });
});
