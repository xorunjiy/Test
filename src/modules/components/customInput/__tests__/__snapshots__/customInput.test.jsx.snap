// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput snapshot should correctly render 1`] = `
<div
  className=""
  display="block"
>
  <input
    className=""
    fontSize={18}
    height={25}
    required={false}
    type="text"
    value=""
    width={40}
  />
</div>
`;
